File: freight_management/freight_management/report/vessel_summary/vessel_summary.py
# Copyright (c) 2025, Shreyas and contributors
# For license information, please see license.txt


import frappe
from frappe import _
from frappe.utils import getdate

def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data

def get_columns():
    return [
        {"label": _("Vessel Name"), "fieldname": "name1", "fieldtype": "Data", "width": 150},
        {"label": _("Shipping Line"), "fieldname": "shipping_line", "fieldtype": "Link", "options": "Shipping Line", "width": 120},
        {"label": _("Draft (m)"), "fieldname": "draft", "fieldtype": "Float", "width": 100},
        {"label": _("LOA"), "fieldname": "loa", "fieldtype": "Float", "width": 100},
        {"label": _("GRT"), "fieldname": "grt", "fieldtype": "Float", "width": 100},
        {"label": _("ETA"), "fieldname": "eta", "fieldtype": "Datetime", "width": 140},
        {"label": _("ETB"), "fieldname": "etb", "fieldtype": "Datetime", "width": 140},
        {"label": _("ETD"), "fieldname": "etd", "fieldtype": "Datetime", "width": 140},
        {"label": _("Discharge 20'"), "fieldname": "discharge_20", "fieldtype": "Int", "width": 110},
        {"label": _("Discharge 40'"), "fieldname": "discharge_40", "fieldtype": "Int", "width": 110},
        {"label": _("Load 20'"), "fieldname": "load_20", "fieldtype": "Int", "width": 100},
        {"label": _("Load 40'"), "fieldname": "load_40", "fieldtype": "Int", "width": 100},
        {"label": _("Remarks"), "fieldname": "remarks", "fieldtype": "Data", "width": 200}
    ]

def get_data(filters):
    conditions = []
    if filters:
        if filters.get("from_date"):
            conditions.append(["eta", ">=", getdate(filters["from_date"])])
        if filters.get("to_date"):
            conditions.append(["eta", "<=", getdate(filters["to_date"])])
        if filters.get("shipping_line"):
            conditions.append(["shipping_line", "=", filters["shipping_line"]])

    return frappe.get_all("Vessels",
        filters=conditions,
        fields=[
            "name1", "shipping_line", "draft", "loa", "grt",
            "eta", "etb", "etd", "discharge_20", "discharge_40",
            "load_20", "load_40", "remarks"
        ]
    )

File: freight_management/freight_management/report/vessel_summary/vessel_summary.js
// Copyright (c) 2025, Shreyas and contributors
// For license information, please see license.txt

frappe.query_reports["Vessel Summary"] = {
	filters: [
        {
            fieldname: "from_date",
            label: "From Date",
            fieldtype: "Date",
            default: frappe.datetime.month_start()
        },
        {
            fieldname: "to_date",
            label: "To Date",
            fieldtype: "Date",
            default: frappe.datetime.month_end()
        },
        {
            fieldname: "shipping_line",
            label: "Shipping Line",
            fieldtype: "Link",
            options: "Shipping Line"
        }
    ],
	formatter: function(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);
        if (column.fieldname === "name1" && data && data.name) {
            return `<a href="/app/vessels/${data.name}" style="font-weight:bold">${value}</a>`;
        }
        return value;
    }
};
